src/syncemailclient.cpp

```diff
@@ -86,8 +80,15 @@ bool SyncEmailClient::uninit()
 
 bool SyncEmailClient::startSync()
 {
-    qDebug() << Q_FUNC_INFO << "Starting scheduled sync for email account: " << m_accountId.toULongLong();
-    m_emailAgent->syncAccounts(QMailAccountIdList() << m_accountId);
+    if (m_emailAgent->ipcConnected()) {
+        triggerSync();
+    } else {
+        qWarning() << Q_FUNC_INFO << "IPC not connect yet, waiting....";
+        connect(m_emailAgent, SIGNAL(ipcConnectionEstablished()), this, SLOT(ipcConnected()));
+        // Since the process running this plugin can cause a wakelock we dont want to wait long time for IPC connection
+        connect(&m_ipcTimeout, SIGNAL(timeout()), this, SLOT(ipcTimeout()));
```

Comment:
You mean ipc connected signals ? this plugin should run only once and exit(runs "inside" buteo not messageserver), so my intention is to listen to this signal only when server is down and for a max of 30 sec, if signal comes we disconnect and after that it should succeeded or fail.